drivers/charger/charger_bq25180.c

```diff
@@ -134,10 +140,68 @@ static int bq25180_get_charge_current(const struct device *dev,
 	return 0;
 }
 
+static int bq25180_get_online(const struct device *dev,
+			      enum charger_online *online)
+{
+	const struct bq25180_config *cfg = dev->config;
+	uint8_t val;
+	int ret;
+
+	ret = i2c_reg_read_byte_dt(&cfg->i2c, BQ25180_STAT0, &val);
+	if (ret < 0) {
+		return ret;
+	}
+
+	if ((val & BQ25180_STAT0_VIN_PGOOD_STAT) != 0x00) {
```

Comment:
`if (val & BQ25180_STAT0_VIN_PGOOD_STAT)` is a bit easier to read

Reply:
I very much agree but formally we should stick with these: https://docs.zephyrproject.org/latest/contribute/coding_guidelines/index.html, "Rule 14.4" in particula: https://gitlab.com/MISRA/MISRA-C/MISRA-C-2012/Example-Suite/-/blob/master/R_14_04.c.

I'll spare you my opinion about those rules, but that specific one has been pointed out to me in the past few times so I try to stick to it these days.

Reply:
got it, thanks for the reference!